packages/core/src/Game/World.js

```diff
@@ -0,0 +1,429 @@
+   */
+  on(eventID, cb) {
+    if (!this.listeners[eventID]) this.listeners[eventID] = [];
+    this.listeners[eventID].push(cb);
+  }
+
+  /**
+   * Dispatch custom event to listeners
+   *
+   * @param {string} eventID - Id of the event to dispatch
+   * @param {Array} params - Params to passed to listeners
+   */
+  dispatch(eventID, params) {
+    if (!this.listeners[eventID]) {
+      console.warn('no listener on event ', eventID);
+    } else {
+      this.listeners[eventID].forEach(function (cb) {
+        cb(params);
+      });
+    }
+  }
+
+  /**
+   * Load its gameobject
+   *
+   * @param {Function} onLoad callback called at the end of the load
+   * @param {WorldContext} worldContext world context to initialize the gameobject
+   */
+  load(onLoad, worldContext) {
+    this.addGameObject(this.gameObject, worldContext, null, onLoad);
+  }
+
+  /**
+   * Compute all the promises of a gameobject needed at the load event of WorldScripts
+   *
+   * @param {GameObject} go the gameobject to compute load promises
+   * @returns {Array[Promise]} An array containing all the promises
+   */
+  computePromisesLoad(go) {
+    const promises = [];
+
+    go.traverse(function (g) {
+      const scriptC = g.getComponent(WorldScript.Model.TYPE);
+      if (scriptC) {
+        const scripts = scriptC.getController().getScripts();
+        for (const idScript in scripts) {
+          const result = scriptC
+            .getController()
+            .executeScript(
+              scripts[idScript],
+              WorldScript.Controller.EVENT.LOAD,
+              []
+            );
+          if (result) promises.push(result);
+        }
+      }
+    });
+
+    return promises;
+  }
+
+  /**
+   * It will dispatch an event to all the world scripts in the gameobject
+   *
+   * @param {GameObject} gameObject - The gameobject that you want to dispatch the event to.
+   * @param {string} event - The name of the event to dispatch @see {WorldScript.Controller.EVENT}.
+   * @param {Array} params - The params to pass to the {@link WorldScriptController} @see {WorldScript.Controller}.
+   */
+  dispatchWorldScriptEvent(gameObject, event, params = []) {
+    gameObject.traverse(function (child) {
+      const worldScriptGameComponent = child.getComponent(
+        WorldScript.Model.TYPE
+      );
+      if (worldScriptGameComponent) {
+        worldScriptGameComponent.getController().execute(event, params);
+      }
+    });
+  }
+
+  /**
+   * Add a GameObject into this world
+   * Then call a callback onLoad (optional)
+   *
+   * @param {GameObject} gameObject - The gameobject to add
+   * @param {WorldContext} worldContext - The world context to initialize the gameobject
+   * @param {GameObject} parent - The gameobject parent may be null
+   * @param {Function} onLoad - Callback called when loaded (optional)
+   */
+  addGameObject(gameObject, worldContext, parent, onLoad = null) {
+    const _this = this;
+
+    worldContext
```

Comment:
remove param worldcontext and do the init in worldcontext class

Reply:
worldcontext should do operation on go.controller => same structure as browsercontext
worldstatecomputer could a component process ticker something like that